osu.Game/Screens/Edit/Compose/Components/SelectionHandler.cs

```diff
@@ -421,7 +412,11 @@ private void createStateBindables()
 
             // bring in updates from selection changes
             EditorBeatmap.HitObjectUpdated += _ => UpdateTernaryStates();
-            EditorBeatmap.SelectedHitObjects.CollectionChanged += (sender, args) => UpdateTernaryStates();
+            EditorBeatmap.SelectedHitObjects.CollectionChanged += (sender, args) =>
+            {
+                Scheduler.AddOnce(updateVisibility);
+                UpdateTernaryStates();
```

Comment:
this could probably also be an `AddOnce`, but maybe that's best for a separate change..